libbeat/outputs/elasticsearch/client.go

```diff
@@ -178,6 +177,25 @@ func eventBulkMeta(index string, event common.MapStr) bulkMeta {
 	return meta
 }
 
+// getIndex returns the full index name
+// Index is either defined in the config as part of the output
+// or can be overload by the event through setting index
+func getIndex(event common.MapStr, index string) string {
+
+	ts := time.Time(event["@timestamp"].(common.Time)).UTC()
+
+	// Check if index is ste dynamically
+	if dynamicIndex, ok := event["index"]; ok {
```

Comment:
good idea. We should probably to the same for type. An other idea I had was putting it under the namespace beat, so it will never conflict. In case we go with an event object (instead of MapStr) it could also be meta-information stored in the event object.